### PR TITLE
Feature/support cold rx client

### PR DESCRIPTION
**Pull request by simpleAndrew**
Changed implementation of RxRestClient to produce "cold" observables

### CODE REVIEW

**Comment by simpleAndrew** [comment]
I didn't change the version of this lib, but maybe it makes sense.

File: src/main/java/com/hubrick/vertx/rest/rx/impl/DefaultRxRestClient.java [hunk 1]
```diff
@@ -30,73 +32,122 @@
...
+
+        final RestClientRequest<T> request = restClient.request(method, uri, responseClass, handler)
+                        .exceptionHandler(errorHandler);
 
-        final RestClientRequest<T> originalRequest = restClient.request(method, uri, responseClass, handler);
-        originalRequest.exceptionHandler(event -> handler.fail(event));
-        final RestClientRequest<T> request = new DefaultRxRestClientRequest<>(originalRequest);
+        final Runnable requestExecution = () -> buildRequestExecutor(requestBuilder, errorHandler, request);
+
+        if (isHot) {
```
Comment (eemmiirr) [inline src/main/java/com/hubrick/vertx/rest/rx/impl/DefaultRxRestClient.java]: I don't think this is needed make it cold and keep it like that.

File: src/main/java/com/hubrick/vertx/rest/rx/impl/DefaultRxRestClient.java [hunk 2]
```diff
@@ -30,73 +32,122 @@
...
-        // We assume builder will call request.end()
+        return Observable.create(new ColdDecorator<>(handler.getSubscribe(), requestExecution));
+    }
+
+    /**
+     * Use the builder to create the full request (or start upload)
+     * We assume builder will call request.end() and trigger request execution.
+     * Request is wrapped in Rx decorator to prevent handler override by requestBuilder.
+     */
+    private <T> void buildRequestExecutor(Action1<RestClientRequest<T>> requestBuilder, Handler<Throwable> errorHandler, RestClientRequest<T> request) {
```
Comment (eemmiirr) [inline src/main/java/com/hubrick/vertx/rest/rx/impl/DefaultRxRestClient.java]: Executors are usually refering to something different in java. Can we have another name?

**Comment by simpleAndrew** [comment]
Just ignore that fact there're 2 commits - I'll squash them before merging the branch with commit message "Convert DefaultRxRestClient into ColdRxRestClient"

File: src/main/java/com/hubrick/vertx/rest/rx/impl/ColdRxRestClient.java [hunk 3]
```diff
@@ -20,83 +20,92 @@
...
+import rx.Emitter;
 import rx.Observable;
 import rx.functions.Action1;
 
 /**
  * @author Emir Dizdarevic
  * @since 1.1.0
  */
-public class DefaultRxRestClient implements RxRestClient {
+public class ColdRxRestClient implements RxRestClient {
```
Comment (megglos) [inline src/main/java/com/hubrick/vertx/rest/rx/impl/ColdRxRestClient.java]: as is this a breaking change already maybe let's switch to single as well?

Comment (simpleAndrew) [reply src/main/java/com/hubrick/vertx/rest/rx/impl/ColdRxRestClient.java]: Will create a separate PR for this soon then

**Pull request merged**